lib/spack/docs/tutorial_buildsystems.rst

```diff
@@ -0,0 +1,499 @@
+can utilize some of Spack's helper functions to customize the building and 
+installing of a package.
+
+-------------------
+Autotools
+-------------------
+
+As we have seen earlier, packages using :code:`Autotools` use :code:`configure`, 
+:code:`make` and :code:`make install` commands to execute the build and 
+install process. In our :code:`Package` class, your typical build incantation will 
+consist of the following:
+
+.. code-block:: python
+
+    def install(self, spec, prefix):
+        configure("--prefix=" + prefix)
+        make()
+        make("install")
+
+You'll see that this looks similar to what we wrote in our packaging tutorial.
+
+The :code:`Autotools` subclass aims to simplify writing package files and provides
+convenience methods to manipulate each of the different phases for a :code:`Autotools`
+build system.
+
+:code:`Autotools` packages consist of four phases:
+
+1. :code:`autoreconf()`
+2. :code:`configure()`
+3. :code:`build()`
+4. :code:`install()`
+
+
+Each of these phases have sensible defaults. Let's take a quick look at some 
+the internals of the :code:`Autotools` class:
+
+.. note:: 
+    The examples showing code for these classes is abridged to avoid having
+    long examples. We only show what is relevant to the packager.
+
+
+.. literalinclude:: tutorial/examples/Autotools/autotools_class.py
+    :language: python
+    :emphasize-lines: 42,45,62
+    :lines: 40-95,259-267
+    :linenos:
+
+
+Important to note are the highlighted lines. These properties allow the 
+packager to set what build targets and install targets they want for their 
+package. If, for example, we wanted to add as our build target :code:`foo` 
+then we can append to our :code:`build_targets` property:
+
+.. code-block:: python
+
+    build_targets = ["foo"]
+
+Which is similiar to invoking make in our Package 
+
+.. code-block:: python
+    
+     make("foo")
+
+Another thing we have highlighted is in the :code:`configure()` method. 
+Here we see that the prefix option is already included which reduces 
+repeated code. We can override `configure_args()` and the output of that
+function will be passed to `configure()` which means that we shouldn't have
+to override `configure` but rather simply construct arguments in `configure_args()`.
+
+Let's look at the mpileaks package.py file that we worked on earlier:
+
+
+.. code-block:: console
+    
+    $ spack edit mpileaks
+
+Notice that mpileaks is a :code:`Package` class but uses the :code:`Autotools` 
+build system. Although this package is acceptable let's make this into an 
+:code:`AutotoolsPackage` class and simplify it further.
+
+.. literalinclude:: tutorial/examples/Autotools/0.package.py
+   :language: python
+   :emphasize-lines: 28
+   :linenos:
+
+We first inherit from the :code:`AutotoolsPackage` class. 
+
+We can keep the install method but we can simplify this further by adding 
+configure_args:
```

Comment:
I would replace this paragraph with:

"Although we could keep the `install` method, most of it can be handled by the `AutotoolsPackage` base class. In fact, the only thing that needs to be overridden is `configure_args`:"